### PR TITLE
Cygwin fixes

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This PR fixes two problems which stop NormalizInterface working in Cygwin/Windows.

1) Switch to using gnu++11 instead of c++11. This is because GAP's configure script it built using the default compiler settings, which enable gnu extensions. I don't know if this could break another OS, but the same switch was made in Semigroups and doesn't seem to have caused problems.

2) Move the normaliz DLL to gaproot/.libs. This does look horrible, but in windows I can use ProcessMonitor and this is the only place inside GAPROOT which is checked for this file. We could alternatively install it globall, in /usr/lib say (still inside the Cygwin root obviously).

With these two changes, the package works fine in Cygwin, and passes testall.g

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -4,7 +4,7 @@
 KEXT_NAME = NormalizInterface
 KEXT_SOURCES = src/normaliz.cc
 
-KEXT_CXXFLAGS = @NORMALIZ_CPPFLAGS@ -std=c++11
+KEXT_CXXFLAGS = @NORMALIZ_CPPFLAGS@ -std=gnu++11
```
Comment (fingolfin) [inline Makefile.in]: Hmm, so the problem there was that sysjmp.h or so uses a GNU C extension, right? I am OK with adding a workaround to make sure things work with GAP 4.11.0 but. But I'd really prefer to fix the root cause on the GAP side, and make sure GAP headers are usable without GNU extensions. That shouldn't be too hard to fix thanks to our recent header restructuring -- at least on `master`, but hopefully we can achieve that on `stable-4.11`, too. I'll try to work on a PR for GAP later (have to take care of the kid right now, no babysitter today)

File: build-normaliz.sh [hunk 2]
```diff
@@ -116,3 +116,12 @@ else
 fi
 make -j4
 make install
+
+if [ "$(expr substr $(uname -s) 1 6)" == "CYGWIN" ]; then
```
Comment (fingolfin) [inline build-normaliz.sh]: Very minor point: I am not sure that is POSIX portable? The script should not break here if run in e.g. dash or zsh. I can suggest a different check later (when I am at my laptop).

File: build-normaliz.sh [hunk 3]
```diff
@@ -116,3 +116,12 @@ else
...
+    echo "##"
+    echo "## Extra Cygwin installation step"
+    echo "##"
+    # We have to move the Normalize dll to $GAPDIR/.libs, as this is the only
+    # place which Cygwin will check for it inside $GAPDIR.
+    cp ${NormalizInstallDir}/bin/cygnormaliz-*.dll "${GAPDIR}/.libs"
```
Comment (fingolfin) [inline build-normaliz.sh]: I am really confused by this. First off, won't this break for users who don't have write access to the GAPDIR?

Secondly, why is this needed here but not for other packages with kernel extensions? This is really weird.

Comment (fingolfin) [reply build-normaliz.sh]: I mean for other packages with kernel extensions that link against external libraries -- such as semigroups and float.

I'd really prefer to understand this and deal with it properly. On other operating systems, a shared library  or a loadable module can contain rpaths that allow specifying locations where to look for further shared libraries. Apparently this is functionality is either missing, or our code to use it is broken on Cygwin and/or Windows. If it is a former, a reference stating that would be nice; but I am kinda hoping for the latter, and then we "just" have to adopt that (and it may requires changes to the GAP build system, etc., but on the long term that's something we have to deal with anyway).

Comment (ChrisJefferson) [reply build-normaliz.sh]: This will break for users who can't write to GAPDIR. However, I would expect there to be very few of those using Cygwin.

This is only necessary for semigroups and normaliz. The problem is that you make a DLL kernel extension, which itself tries to load a second DLL. It's the second DLL which causes problems, as Windows/Cygwin don't know where to look for it. They do not check the directory the first DLL is in.

A better windows expert might know how a better way to fix this -- I just looked for places which I could see Windows was looking in, and picked the "best" one.

Comment (ChrisJefferson) [reply build-normaliz.sh]: So, float doesn't build on Windows generally, Semigroups already has similar code in it. That's not to say we shouldn't do something "nicer".

Comment (ChrisJefferson) [reply build-normaliz.sh]: Just to give some background (this is what I found when looking earlier)

https://cmake.org/pipermail/cmake/2007-October/017268.html

So we could look at putting more directories in the path, or requiring these DLLs are installed "globally" in Cygwin (into /usr/lib). Those seem to be the most sensible options.

File: Makefile.in (hunk 1 above)
Comment (ChrisJefferson) [reply Makefile.in]: To be concrete, the problem is that with extensions turned on, Cygwin's gcc has siglongjmp and sigjmp_buf, so the configure script enables HAS_SIGSETJMP, which is used in system.h

However, then if we compile with extensions, the compiliation fails due to these things being missing.

Comment (fingolfin) [reply Makefile.in]: OK, here is what's going on: the problem is not that GNU extensions are disabled, the problem is that POSIX extensions are *also* disabled. This suggests a way to fix this on the GAP side. Can you please try if this also works:

```suggestion
KEXT_CXXFLAGS = @NORMALIZ_CPPFLAGS@ -std=c++11 -D _POSIX_C_SOURCE=1
```

If it does, that would suggest a way to fix this in GAP: we just add something like this in `sysjmp.h` (resp. in `system.h`, for GAP 4.11) before the `#include <setjmp.h>`:
```C
#ifndef _POSIX_C_SOURCE
#define _POSIX_C_SOURCE 1
#endif
#include <setjmp.h>
```

Comment (ChrisJefferson) [reply Makefile.in]: Check everything over and this seems to fix the problem, and is the only problem in headers.

File: build-normaliz.sh (hunk 3 above)
Comment (fingolfin) [reply build-normaliz.sh:126]: Thank you for all the explanations. While I don't "like" it, it seems like the pragmatic thing to do.

File: Makefile.in (hunk 1 above)
Comment (fingolfin) [inline Makefile.in:7]:
```suggestion
# workaround for GAP 4.11: we need _POSIX_C_SOURCE=1 to avoid an error during compilation;
# see https://github.com/gap-packages/NormalizInterface/pull/91 for details
KEXT_CXXFLAGS = @NORMALIZ_CPPFLAGS@ -std=c++11 -D_POSIX_C_SOURCE=1
```

File: build-normaliz.sh (hunk 2 above)
Comment (fingolfin) [reply build-normaliz.sh:120]: This one ought to work in all POSIX compatible shells:
```suggestion
osname=$(uname -s)
if [ "${osname#*$CYGWIN}" != "$osname" ]; then
```

File: Makefile.in (hunk 1 above)
Comment (fingolfin) [reply Makefile.in]: Unfortunately defining _POSIX_C_SOURCE like this causes a build failure for this package on FreeBSD. Luckily in 4.12 this will all be gone due to my restructuring of the GAP header files.

Comment (fingolfin) [reply Makefile.in]: Due to this, I am going back to using `-std=gnu++11` (and once GAP 4.12 is out, we can just depend on that instead)
